### PR TITLE
add qiskit 0.42.0 configuration files

### PR DESCRIPTION
**Pull request by MoiseRousseau**
Add qiskit-0.42.0 configuration files and update some of its dependencies.

### CODE REVIEW

File: config/rustworkx.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+MODULE_BUILD_DEPS="python-build-bundle rust"
```
Comment (ccoulombe) [inline config/rustworkx.sh:1]: The `python-build-bundle` is automatically present, you can in fact remove this dep

File: config/qiskit-aer-0.12.0.sh
Comment (ccoulombe) [inline config/qiskit-aer-0.12.0.sh:1]: Does it need to only for 0.12.0 version, or is it general enough ? It looks general to me, hence you could rename the file for `qiskit-aer.sh`

File: config/retworkx.sh [hunk 2]
```diff
@@ -1,2 +1 @@
-PYTHON_DEPS="setuptools-rust"
-MODULE_RUNTIME_DEPS="rust/1.59.0"
+MODULE_BUILD_DEPS="python-build-bundle rust"
```
Comment (ccoulombe) [inline config/retworkx.sh:1]: The python-build-bundle is automatically present, you can in fact remove this dep

**Review by ccoulombe — CHANGES_REQUESTED** [review]

**Comment by MoiseRousseau** [comment]
All suggestions integrated!

**Pull request merged**